packages/next/build/webpack-config.ts

```diff
@@ -1024,6 +1025,23 @@ export default async function getBaseWebpackConfig(
       webpack,
     })
 
+    if (
+      dev &&
+      webpackConfig.devtool &&
+      originalDevtool !== webpackConfig.devtool
+    ) {
+      webpackConfig.devtool = originalDevtool
+
+      // only show warning for one build
+      if (isServer) {
```

Comment:
What if the user only overrode client? They'd never see the message.

Our `execOnce` should be able to help with this.